File: Data Science and Machine Learning Essentials/Lab2D Data Cleansing and Transformation in Azure ML/DAT203x_Lab02D/VisualizeOutliers.R
# Map 1-based optional input ports to variables
dataset1 <- maml.mapInputPort(1) # class: data.frame

## remove columns
cols <- c("X", "Y", "month", "day")
dataset1 <- dataset1[, !(names(dataset1)) %in% cols]

## Create a pairs plot. 
pairs(dataset1)

File: Data Science and Machine Learning Essentials/Lab2D Data Cleansing and Transformation in Azure ML/DAT203x_Lab02D/VisualizeOutliers.py
def azureml_main(frame1):
  import matplotlib
  matplotlib.use('agg')
  
  import pandas as pd
  import matplotlib.pyplot as plt
  from pandas.tools.plotting import scatter_matrix 

## Remove unwanted columns 
  frame1.drop(["X", "Y", "month", "day"], axis = 1, inplace = True)

## Create a scatter plot matrix 
  fig1 = plt.figure(1, figsize = (12,9))
  ax = fig1.gca()
  scatter_matrix(frame1, alpha=0.2, figsize=(10, 10), diagonal='kde', ax=ax)
  fig1.savefig('scatter2.png')
  return frame1
